clients/da-vinci-client/src/main/java/com/linkedin/davinci/consumer/VeniceChangelogConsumerImpl.java

```diff
@@ -142,13 +143,21 @@ public VeniceChangelogConsumerImpl(
         changelogClientConfig.getInnerClientConfig(),
         VeniceProperties.empty(),
         null);
-    recordChangeEventSchemaRepository = new RecordChangeEventReadOnlySchemaRepository(this.storeRepository);
-    Class<V> valueClass = changelogClientConfig.getInnerClientConfig().getSpecificValueClass();
-    if (valueClass != null) {
+    this.recordChangeEventDeserializerCache = new AvroStoreDeserializerCache<>(
+        new RecordChangeEventReadOnlySchemaRepository(this.storeRepository),
+        storeName,
+        true);
+    if (changelogClientConfig.getInnerClientConfig().isSpecificClient()) {
       // If a value class is supplied, we'll use a Specific record adapter
-      userEventChunkingAdapter = new SpecificRecordChunkingAdapter(valueClass);
+      Class valueClass = changelogClientConfig.getInnerClientConfig().getSpecificValueClass();
+      this.userEventChunkingAdapter = new SpecificRecordChunkingAdapter();
+      this.storeDeserializerCache = new AvroStoreDeserializerCache<>(
+          id -> storeRepository.getValueSchema(storeName, id).getSchema(),
+          (writerSchema, readerSchema) -> FastSerializerDeserializerFactory
+              .getFastAvroSpecificDeserializer(writerSchema, valueClass));
```

Comment:
You're absolutely right! Thanks for catching this! I also cleaned up the `AvroStoreDeserializerCache` constructors a bit, since some of them were unnecessary.